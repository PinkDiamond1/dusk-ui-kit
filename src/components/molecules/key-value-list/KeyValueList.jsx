/* @flow */
import * as React from 'react'

import KeyValueListItem from '../../atoms/key-value-list-item'

type Props = {
  children: React.Node,
  className: string,
}

const KeyValueList = (props: Props): React.Element<*> => {
  const { children, className } = props
  const updatedClassName = `__duk-key-value-list row ${className}`
  return <dl className={updatedClassName}>{children}</dl>
}

KeyValueList.defaultProps = {
  className: '',
}

KeyValueList.Item = KeyValueListItem

export default KeyValueList
