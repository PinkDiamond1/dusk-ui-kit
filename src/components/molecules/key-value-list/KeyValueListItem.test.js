import React from 'react'
import renderer from 'react-test-renderer'

import KeyValueList from './KeyValueList'

it('test to see if the KeyValueList renders correctly', () => {
  const tree = renderer.create(<KeyValueList>Smoke Test</KeyValueList>).toJSON()
  expect(tree).toMatchSnapshot()
})
