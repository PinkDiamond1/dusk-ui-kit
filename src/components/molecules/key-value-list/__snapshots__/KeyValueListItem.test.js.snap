// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test to see if the KeyValueList renders correctly 1`] = `
<dl
  className="__duk-key-value-list "
>
  Smoke Test
</dl>
`;
